src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -364,6 +365,26 @@ public bool IsSetAsOutput
             }
         }
 
+        /// <summary>
+        /// This property defines if the node is a PythonNode
+        /// </summary>
+        [JsonIgnore]
+        internal bool IsPython
+        {
+            get
+            {
+                return isPython;
+            }
+
+            set
+            {
+                if (isPython != value)
+                {
+                    isPython = value;
+                }
+            }
+        }
+
```

Comment:
I say get rid of this new property and rather add a new internal virtual method `GetNameOfAssemblyReferencedByNode`
It is already specialized for DSFunction, you could do the same for PythonNodeBase (return the assembly location of the PythoNEngine type that corresponds with the currently selected engine name)

Reply:
So...add PythonNodeModels dependency to Core?

Reply:
No. You can just override the base method in whatever class needs to customize it. Like DSFunction and PythonNodeModelBase

Reply:
Ok, but where should this new internal virtual method be? in `WorkspaceModel`?

Reply:
NodeModel